int32 id
bool terminate
string cause